ui/pages/confirmations/components/smart-transactions-banner-alert/smart-transactions-banner-alert.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { TransactionType } from '@metamask/transaction-controller';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  BannerAlert,
+  ButtonLink,
+  Text,
+  BannerAlertSeverity,
+} from '../../../../components/component-library';
+import { setAlertEnabledness } from '../../../../store/actions';
+import { AlertTypes } from '../../../../../shared/constants/alerts';
+import { SMART_TRANSACTIONS_LEARN_MORE_URL } from '../../../../../shared/constants/smartTransactions';
+import { FontWeight } from '../../../../helpers/constants/design-system';
+import { useConfirmContext } from '../../context/confirm';
+import { isCorrectDeveloperTransactionType } from '../../../../../shared/lib/confirmation.utils';
+import {
+  getSmartTransactionsOptInStatusInternal,
+  getSmartTransactionsMigrationAppliedInternal,
+} from '../../../../../shared/modules/selectors/smart-transactions';
+
+type MarginType = 'default' | 'none' | 'noTop' | 'onlyTop';
+
+type SmartTransactionsBannerAlertProps = {
+  marginType?: MarginType;
+};
+
+export const SmartTransactionsBannerAlert: React.FC<SmartTransactionsBannerAlertProps> =
+  React.memo(({ marginType = 'default' }) => {
+    const t = useI18nContext();
+
+    let currentConfirmation;
+    try {
+      const context = useConfirmContext();
+      currentConfirmation = context?.currentConfirmation;
+    } catch {
+      currentConfirmation = null;
+    }
+
+    const alertEnabled = useSelector(
+      (state: {
+        metamask: { alertEnabledness?: { [key: string]: boolean } };
+      }) =>
+        state.metamask.alertEnabledness?.[
+          AlertTypes.smartTransactionsMigration
+        ] !== false,
+    );
+
+    const smartTransactionsOptIn = useSelector(
+      getSmartTransactionsOptInStatusInternal,
+    );
+
+    const smartTransactionsMigrationApplied = useSelector(
+      getSmartTransactionsMigrationAppliedInternal,
```

Comment:
nit: can we remove the `Internal` postfix? So it would be: `getSmartTransactionsMigrationApplied`